### DISCUSSION TITLE
Is it possible to customize the locale prefix?

### DISCUSSION THREAD

**Discussion by huskyjp** [amannn/next-intl#1102] [Q&A]
Hey there - I got a simple question that can we rewrite the locale prefix to something else? 

For example if we set `en` and `de` for the locals but want to rewrite the `en` to `us` (since english pathname is `en-US`), is it possible to achieve this behavior within this package?  In short, what we want is `localhost:3000/` for default locale, but `localhost:3000/us` for english locale, not `localhost:3000/en`.

```
import createMiddleware from 'next-intl/middleware';
 
export default createMiddleware({
  // A list of all locales that are supported
  locales: ['en', 'de'],
 
  // Used when no locale matches
  defaultLocale: 'de',

  localePrefix: "as-needed"
});
```

Because we detect the first language from browser header, we can't set like `locales: ['us', 'de']` since browser just tell us `en` value otherwise it redirectes to the defaultLocale. 

I tried to use `pathnames` feature like below but it didn't work: 
```
pathnames = {
  "/": {
    en: "/us",
    de: "/",
  },
```

**Comment by amannn** [comment]
Hey! This feature is discussed in https://github.com/amannn/next-intl/issues/653 and I'm currently working on an implementation in https://github.com/amannn/next-intl/pull/1086. I'll probably share a prerelease in the linked issue to get feedback from users, you might want to subscribe there if you're interested!
